kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/ResolutionResult.java

```diff
@@ -127,15 +126,15 @@ public Optional<KafkaService> kafkaServiceRef(VirtualKafkaCluster cluster) {
      * Get all resolved Filters
      * @return filters
      */
-    public Collection<GenericKubernetesResource> filters() {
+    public Collection<KafkaProtocolFilter> filters() {
         return filters.values();
     }
 
     /**
      * Get the resolved GenericKubernetesResource for a filterRef
      * @return optional containing the resource if resolved, else empty
      */
-    public Optional<GenericKubernetesResource> filter(FilterRef filterRef) {
+    public Optional<KafkaProtocolFilter> filter(FilterRef filterRef) {
```

Comment:
for a future PR: we should change this method to not compare all the components of filterRef. We should convert `filterResource` to an anyLocalRef and compare directly with filterRef.